processing/src/main/java/org/apache/druid/segment/data/FrontCodedIndexedWriter.java

```diff
@@ -337,15 +340,7 @@ public static int unsignedCompare(
     if (b2 == null) {
       return 1;
     }
-    final int commonLength = Math.min(b1.length, b2.length);
-
-    for (int i = 0; i < commonLength; i++) {
-      final int cmp = FrontCodedIndexed.unsignedByteCompare(b1[i], b2[i]);
-      if (cmp != 0) {
-        return cmp;
-      }
-    }
 
-    return Integer.compare(b1.length, b2.length);
+    return StringUtils.compareUtf8(b1, b2);
```

Comment:
I think for now this should probably use `compareUtf8UsingJavaStringOrdering`, additionally, these lines https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/data/FrontCodedIndexed.java#L358 and https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/segment/data/FrontCodedIndexed.java#L406 should also be updated to use `compareUtf8UsingJavaStringOrderingSingleByte`

Reply:
Just made that change. Do you know of a test case that would pass with this change, but fail otherwise? I'll add it, if so.

Reply:
Added: https://github.com/apache/druid/pull/13364/commits/ffd950c4ef525cd025c498e8663319d7fb00b704. And, yes, the order is swapped.